air-quality-backend/tests/api/test_main.py

```diff
@@ -1,11 +1,274 @@
+import mongomock
+import pytest
 from fastapi.testclient import TestClient
-
+from unittest.mock import patch
 from air_quality.api.main import app
+from tests.database.forecast_database_test_data import forecast_from_database
 
 client = TestClient(app)
 
 
-def test_read_main():
-    response = client.get("/")
-    assert response.status_code == 200
-    assert response.json() == {"message": "Hello World"}
+@pytest.fixture
+def mock_collection():
+    yield mongomock.MongoClient(tz_aware=True).db.collection
+
+
+def test_get_forecast_data_no_city_name(mock_collection):
+    with patch(
+        "src.air_quality.database.forecasts.get_collection",
+        return_value=mock_collection,
+    ):
+        mock_collection.insert_many(forecast_from_database)
+
+        valid_date_from = "2024-05-27T12%3A00%3A00.000%2B00%3A00"
+        valid_date_to = "2024-05-27T23%3A00%3A00.000%2B00%3A00"
+        location_type = "city"
+        forecast_base_time = "2024-05-27T12%3A00%3A00.000%2B00%3A00"
+        name = "Abu%20Dhabi"
+        response = client.get(
+            "/air-pollutant-forecast?valid_date_from="
+            + valid_date_from
+            + "&valid_date_to="
+            + valid_date_to
+            + "&location_type="
+            + location_type
+            + "&forecast_base_time="
+            + forecast_base_time
+            + "&location_name="
+            + name
+        )
+        assert response.status_code == 200
+
+
+def test_get_forecast_data_with_city_name(mock_collection):
```

Comment:
I can't see a test in here that checks the 200 response contains what it should. I used cerberus before in another test which was quite nice, but could also be done just asserting as you are everywhere else